hadoop-tools/hadoop-aliyun/src/test/java/org/apache/hadoop/fs/aliyun/oss/TestAliyunOSSRedirect.java

```diff
@@ -0,0 +1,264 @@
+  }
+
+  private void cleanRedirectRule(OSSClient ossClient) {
+    try {
+      // 填写Bucket名称。
+      ossClient.deleteBucketWebsite(bucketName);
+    } catch (OSSException oe) {
+      LOG.error("Caught an OSSException, which means your request made it to OSS, "
+          + "but was rejected with an error response for some reason.");
+      LOG.error("Error Message:" + oe.getErrorMessage());
+      LOG.error("Error Code:" + oe.getErrorCode());
+      LOG.error("Request ID:" + oe.getRequestId());
+      LOG.error("Host ID:" + oe.getHostId());
+      throw oe;
+    } catch (ClientException ce) {
+      LOG.error("Caught an ClientException, which means the client encountered "
+          + "a serious internal problem while trying to communicate with OSS, "
+          + "such as not being able to access the network.");
+      LOG.error("Error Message:" + ce.getMessage());
+      throw ce;
+    }
+  }
+
+  private void setRedirectRule(OSSClient ossClient) {
+
+    try {
+      SetBucketWebsiteRequest request = new SetBucketWebsiteRequest(bucketName);
+      List<RoutingRule> routingRules = new ArrayList<RoutingRule>();
+
+      RoutingRule rule = new RoutingRule();
+      rule.setNumber(1);
+      rule.getCondition().setKeyPrefixEquals("test/redirect_test_");
+      rule.getCondition().setHttpErrorCodeReturnedEquals(404);
+      rule.getRedirect().setRedirectType(RoutingRule.RedirectType.External);
+      rule.getRedirect().setHttpRedirectCode(302);
+      rule.getRedirect().setHostName("apache.github.io");
+      rule.getRedirect().setProtocol(RoutingRule.Protocol.Http);
+      rule.getRedirect().setReplaceKeyWith("hadoop/hadoop-aliyun/tools/hadoop-aliyun/index.html");
+
+      routingRules.add(rule);
+      request.setRoutingRules(routingRules);
+      ossClient.setBucketWebsite(request);
+      LOG.info("setBucketWebsite success");
+    } catch (OSSException oe) {
+      LOG.info("Caught an OSSException, which means your request made it to OSS, "
+          + "but was rejected with an error response for some reason.");
+      LOG.info("Error Message:" + oe.getErrorMessage());
+      LOG.info("Error Code:" + oe.getErrorCode());
+      LOG.info("Request ID:" + oe.getRequestId());
+      LOG.info("Host ID:" + oe.getHostId());
+      throw oe;
+    } catch (ClientException ce) {
+      LOG.error("Caught an ClientException, which means the client encountered "
+          + "a serious internal problem while trying to communicate with OSS, "
+          + "such as not being able to access the network.");
+      LOG.error("Error Message:" + ce.getMessage());
+      throw ce;
+    }
+  }
+
+  @Test
+  public void testRedirectDisable() throws Exception {
+    Configuration conf = new Configuration();
+    conf.setBoolean(REDIRECT_ENABLE_KEY, false);
+    fs = AliyunOSSTestUtils.createTestFileSystem(conf);
+
+    Path srcFilePath = new Path(testRootPath,"redirect_test_src.txt");
+    assertThrows(IOException.class, () -> this.fs.open(srcFilePath));
```

Comment:
so if this doesn't throw, there's actually a new open stream. better to use our LambdaTestUtils.intercept() which is often more informative on failures, and close the stream.

```
intercept(IOException.class, () -> 
   fs.open(srcFilePath).close());
```